primitives/src/json.rs

```diff
@@ -0,0 +1,201 @@
+//! String containing serialized valid JSON.
+//! This string is guaranteed to parse as JSON
+
+#[cfg(not(feature = "std"))]
+use alloc::{
+    borrow::ToOwned,
+    string::{String, ToString},
+    vec::Vec,
+};
+use core::{
+    fmt::{Display, Formatter},
+    str::FromStr,
+};
+#[cfg(feature = "std")]
+use std::{
+    string::{String, ToString},
+    vec::Vec,
+};
+
+use iroha_schema::IntoSchema;
+use parity_scale_codec::{Decode, Encode};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde_json::Value;
+
+#[allow(missing_docs)]
+#[derive(Debug, Clone, Eq, Encode, Decode)]
+pub struct JsonString(pub String);
+
+/// A candidate for a valid `JsonString`.
+/// Is used for generalizing ser/de any types to `JsonString` and vise versa
+#[derive(Serialize, Deserialize, Clone)]
+struct JsonCandidate<T>(T);
+
+impl JsonString {
+    /// Construct [`JsonString`]
+    pub fn new<T: Serialize>(payload: T) -> Self {
+        JsonCandidate(payload).try_into().unwrap()
+    }
```

Comment:
doesn't clippy complain that you have no `# Panics` in the documentation? Also, I don't approve of a panicking constructor, but other may disagree

Reply:
Will try to figure out something

Reply:
What are other options? Calling unwrap in every single case won't be convenient and not sure if it would have any benefits.

Currently we only can fail in the following [case](https://docs.rs/serde_json/latest/serde_json/fn.to_string.html):
> Serialization can fail if T’s implementation of Serialize decides to fail, or if T contains a map with non-string keys.

Reply:
@mversic , @Erigara , yep, I couldn't find any other options